src/test/java/com/sfuentes/ticket/TicketTest.java

```diff
@@ -0,0 +1,75 @@
+package com.sfuentes.ticket;
+
+import com.sfuentes.grocery.Client;
+import com.sfuentes.grocery.Grocery;
+import com.sfuentes.grocery.Item;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TicketTest {
+
+  @Test
+  void ticket_decimalResults_returnChange() {
+    Client client = new Client();
+    client.addItem(new Item("Orange", 82.1d));
+    client.addItem(new Item("Potatoes", 80.1d));
+    client.addItem(new Item("Eggs", 40.3d));
+    Grocery grocery = new Grocery();
+    Ticket ticket = new Ticket(grocery, client, 1_000);
+    ticket.calculateVAT();
+    ticket.calculateTotalToPay();
+    ticket.calculateChange();
+
+    assertEquals(32.4, ticket.getVat());
+    assertEquals(234.9, ticket.getTotalToPay());
+    assertEquals(765.1, ticket.getChange());
```

Comment:
Remember a test only validates a SINGLE task.
Here you are validating 3 methods, I think the confusion starts when you are naming your tests, seems that is not well defined what are you actually testing and then it ended up with multiple validations.

Why this is a problem?
Let's imagine that this test fails, instead of giving you a full insight of what went wrong you will have to come here and see which of the 3 asserts is the one failing, they don't belong to the same scope of the test.
Is not just that you don't have to have multiple assertions, that's not bad per se, but they need to belong to the same scope of the test.

Reply:
Also the same for the other test methods.